python/lsst/daf/butler/cli/utils.py

```diff
@@ -65,3 +130,82 @@ def to_upper(context, param, value):
         A copy of the passed-in value, converted to upper case.
     """
     return value.upper()
+
+
+def printFunctionInfo(func, *args, **kwargs):
+    """For unit testing butler subcommand call execution, write a dict to
+    stdout that formats information about a funciton call into a dict that can
+    be evaluated by `verifyFunctionInfo`
+
+    Parameters
+    ----------
+    func : function
+        The function that has been called, whose name should be written.
+    args : [`str`]
+        The values of the arguments the function was called with.
+    kwags : `dict` [`str`, `str`]
+        The names and values of the kwargs the function was called with.
+    """
+    print(dict(function=func.__name__,
+               args=args,
+               kwargs=kwargs))
+
+
+def verifyFunctionInfo(testSuite, output, function, expectedArgs, expectedKwargs):
+    """For unit testing butler subcommand call execution, compare a dict that
+    has been printed to stdout to expected data.
+
+    Parameters
+    ----------
+    testSuite : `unittest.Testsuite`
+        The test suite that is executing a unit test.
+    output : `str`
+        The dict that has been printed to stdout. It should be formatted to
+        re-instantiate by calling `eval`.
+    function : `str`
+        The name of the function that was was expected to have been called.
+    expectedArgs : [`str`]
+        The values of the arguments that should have been passed to the
+        function.
+    expectedKwargs : `dict` [`str`, `str`]
+        The names and values of the kwargs that should have been passsed to the
+        funciton.
+    """
+    calledWith = eval(output)
```

Comment:
if we are going to be using `eval` on the output that we generated above, I think it would be safer all around if we made the "print" do a yaml dump and here we did a yaml safe_load.